fix: fix validation error during workspace creation without preset

SasSwart opened the pull request:
closes https://github.com/coder/coder/issues/18430.

Selecting a preset, and then selecting the "None" preset used to set the preset ID to an empty string instead of `undefined`. This sent `""` to the backend, which expects a valid UUID.

SasSwart committed:
fix: allow delesecting a preset without triggering a validation error

SasSwart committed:
make fmt

dannykopping reviewed (approved):
Not my area so will need FE approval before merging, but seems fine to me
Thanks for the fix 👍

jaaydenh committed:
fix: implement fix for dynamic parameters

ssncferreira reviewed (approved):
LGTM great test addition 🌟

mafredri reviewed (approved):
LGTM 👍🏻

SasSwart committed:
Update site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx

use option?.value instead of index to check whether we should send a preset ID when a workspace is created

Co-authored-by: Susana Ferreira <susana@coder.com>
